src/main/java/seedu/duke/parser/Parser.java

```diff
@@ -263,10 +269,59 @@ public static void parseEditStep(RecipeList recipeList,
             throw new Exception("error in edit step:\n" + e.getMessage());
         }
     }
-    public static String removeForbiddenChars(String ingredient) {
-        for (String chara : StringLib.FORBIDDEN_CHARS) {
-            ingredient.replaceAll(chara, "");
+    public static String[] parseAddToRecipeDescription(String description) {
+        String[] subDescriptions = description.split("/", 3);
+        String element0 = subDescriptions[0].toLowerCase().split("id")[0].trim();
+        subDescriptions[0] = element0.replace("-","");
+        String element1 = subDescriptions[1].toLowerCase().split("desc")[0].trim();
+        subDescriptions[1] = element1;
+        return subDescriptions;
+    }
+    public static String[] parseDeleteFromRecipeDescription(String description) {
+        String[] subDescriptions = description.split("/", 2);
+        String element0 = subDescriptions[0].toLowerCase().split("id")[0].trim();
+        subDescriptions[0] = element0.replace("-","");
+        return subDescriptions;
+    }
+    public static boolean isValidAddToRecipe(String description) {
+        String descLowerCase = description.toLowerCase().trim();
+        if ((descLowerCase.contains("step") ||
+                descLowerCase.contains("ingredient")) &&
+                descLowerCase.contains("id/") &&
+                descLowerCase.contains("desc/") &&
+                !(descLowerCase.contains("step") &&
+                        descLowerCase.contains("ingredient"))) {
```

Comment:
@YatPang Consider the case of people passing in `addtorecipe step id/ 1 desc/ step` or `addtorecipe ingredient id/ 1 desc/ ingredient`? Will this command still be deemed valid?

Reply:
Will be deemed valid. " " in "id/ [n]", "desc/ [step/ingredient]" removed during processing in parseAddToRecipeDescription (Parser:270)